### PR TITLE
Specify UTF-8 when reading build log

### PR DESCRIPTION
**Pull request by elharo**
also buffer the reader

### CODE REVIEW

**Comment by elharo** [comment]
Looks like Mac builds are borked:

  Error: Could not find satisfied version for SemVer '8'. 
  Available versions: 23.0.1+11, 23.0.0+37, 22.0.2+9, 22.0.1+8, 22.0.0+36, 21.0.5+11.0.LTS, 21.0.4+7.0.LTS, 21.0.3+9.0.LTS, 21.0.2+13.0.LTS, 21.0.1+12.0.LTS, 21.0.0+35.0.LTS, 20.0.2+9, 20.0.1+9, 20.0.0+36, 19.0.2+7, 19.0.1+10, 19.0.0+36, 18.0.2+101, 18.0.2+9, 18.0.1+10, 18.0.0+36, 17.0.13+11, 17.0.12+7, 17.0.11+9, 17.0.10+7, 17.0.9+9, 17.0.8+101, 17.0.8+7, 17.0.7+7, 17.0.6+10, 17.0.5+8, 17.0.4+101, 17.0.4+8, 17.0.3+7, 17.0.2+8, 17.0.1+12, 17.0.0+35, 11.0.25+9, 11.0.24+8, 11.0.23+9, 11.0.22+7.1, 11.0.22+7, 11.0.21+9, 11.0.20+101, 11.0.20+8, 11.0.19+7, 11.0.18+10, 11.0.17+8, 11.0.16+101, 11.0.16+8, 11.0.15+10

**Comment by gnodet** [comment]
> Looks like Mac builds are borked:
> 
> Error: Could not find satisfied version for SemVer '8'. Available versions: 23.0.1+11, 23.0.0+37, 22.0.2+9, 22.0.1+8, 22.0.0+36, 21.0.5+11.0.LTS, 21.0.4+7.0.LTS, 21.0.3+9.0.LTS, 21.0.2+13.0.LTS, 21.0.1+12.0.LTS, 21.0.0+35.0.LTS, 20.0.2+9, 20.0.1+9, 20.0.0+36, 19.0.2+7, 19.0.1+10, 19.0.0+36, 18.0.2+101, 18.0.2+9, 18.0.1+10, 18.0.0+36, 17.0.13+11, 17.0.12+7, 17.0.11+9, 17.0.10+7, 17.0.9+9, 17.0.8+101, 17.0.8+7, 17.0.7+7, 17.0.6+10, 17.0.5+8, 17.0.4+101, 17.0.4+8, 17.0.3+7, 17.0.2+8, 17.0.1+12, 17.0.0+35, 11.0.25+9, 11.0.24+8, 11.0.23+9, 11.0.22+7.1, 11.0.22+7, 11.0.21+9, 11.0.20+101, 11.0.20+8, 11.0.19+7, 11.0.18+10, 11.0.17+8, 11.0.16+101, 11.0.16+8, 11.0.15+10

Previous builds were using zulu:
https://github.com/apache/maven-invoker-plugin/actions/runs/12054421435/job/33612825979

**Comment by gnodet** [comment]
> > Looks like Mac builds are borked:
> > Error: Could not find satisfied version for SemVer '8'. Available versions: 23.0.1+11, 23.0.0+37, 22.0.2+9, 22.0.1+8, 22.0.0+36, 21.0.5+11.0.LTS, 21.0.4+7.0.LTS, 21.0.3+9.0.LTS, 21.0.2+13.0.LTS, 21.0.1+12.0.LTS, 21.0.0+35.0.LTS, 20.0.2+9, 20.0.1+9, 20.0.0+36, 19.0.2+7, 19.0.1+10, 19.0.0+36, 18.0.2+101, 18.0.2+9, 18.0.1+10, 18.0.0+36, 17.0.13+11, 17.0.12+7, 17.0.11+9, 17.0.10+7, 17.0.9+9, 17.0.8+101, 17.0.8+7, 17.0.7+7, 17.0.6+10, 17.0.5+8, 17.0.4+101, 17.0.4+8, 17.0.3+7, 17.0.2+8, 17.0.1+12, 17.0.0+35, 11.0.25+9, 11.0.24+8, 11.0.23+9, 11.0.22+7.1, 11.0.22+7, 11.0.21+9, 11.0.20+101, 11.0.20+8, 11.0.19+7, 11.0.18+10, 11.0.17+8, 11.0.16+101, 11.0.16+8, 11.0.15+10
> 
> Previous builds were using zulu: https://github.com/apache/maven-invoker-plugin/actions/runs/12054421435/job/33612825979

The difference is  because you push to branches in this repository.
I do always push to my fork and create a PR from there.

**Comment by slawekjaranowski** [comment]
please rebase branch with current master ...

File: src/main/java/org/apache/maven/plugins/invoker/InvokerSession.java [hunk 1]
```diff
@@ -210,7 +212,8 @@ public void logFailedBuildLog(Log logger, boolean ignoreFailures) throws MojoFai
                     buildLogMessage.append(System.lineSeparator());
                     buildLogMessage.append("*** begin build.log for: " + buildJob.getProject() + " ***");
                     buildLogMessage.append(System.lineSeparator());
-                    try (FileReader buildLogReader = new FileReader(buildLogFile)) {
+                    try (Reader buildLogReader =
+                            Files.newBufferedReader(buildLogFile.toPath(), StandardCharsets.UTF_8)) {
```
Comment (gnodet) [inline src/main/java/org/apache/maven/plugins/invoker/InvokerSession.java:216]: No need to specify the UTF-8 encoding, that's [the default for the NIO File api](https://help.intrexx.com/apidocs/jdk17/api/java.base/java/nio/file/Files.html#newBufferedReader(java.nio.file.Path)).

**Review by gnodet — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/maven/plugins/invoker/InvokerSession.java (hunk 1 above)
Comment (elharo) [reply src/main/java/org/apache/maven/plugins/invoker/InvokerSession.java]: done

**Review by gnodet — APPROVED** [review]

**Pull request merged**